src/main/java/mate/academy/service/impl/BookServiceImpl.java

```diff
@@ -4,22 +4,29 @@
 import lombok.RequiredArgsConstructor;
 import mate.academy.dto.BookDto;
 import mate.academy.dto.CreateBookRequestDto;
+import mate.academy.exception.DuplicateResourceException;
 import mate.academy.exception.EntityNotFoundException;
 import mate.academy.mapper.BookMapper;
 import mate.academy.model.Book;
 import mate.academy.repository.BookRepository;
 import mate.academy.service.BookService;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
 
+    @Transactional
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
+        if (bookRepository.existsByIsbn(book.getIsbn())) {
```

Comment:
remove checking

Reply:
I removed it, but when I tried to add a book with an invalid ISBN DB increments ID and when I used a valid ISBN, DB it use already id+1. What should I do in this case?